src/App.spec.ts

```diff
@@ -308,6 +308,159 @@ describe('App', () => {
+              body: {
+                command: '/COMMAND_NAME',
+              },
+              respond: noop,
+              ack: noop,
+            },
+            { // IncomingEventType.Action (app.action)
+              body: {
+                type: 'block_actions',
+                actions: [{
+                  action_id: 'block_action_id'
+                }],
+                channel: {},
+                user: {},
+                team: {},
+              },
+              respond: noop,
+              ack: noop,
+            },
+            { // IncomingEventType.Action (app.action)
+              body: {
+                type: 'message_action',
+                callback_id: 'message_action_callback_id',
+                channel: {},
+                user: {},
+                team: {},
+              },
+              respond: noop,
+              ack: noop,
+            },
+            { // IncomingEventType.Action (app.action)
+              body: {
+                type: 'interactive_message',
+                callback_id: 'interactive_message_callback_id',
+                actions: [{}],
+                channel: {},
+                user: {},
+                team: {},
+              },
+              respond: noop,
+              ack: noop,
+            },
+            { // IncomingEventType.Action with dialog submission (app.action)
+              body: {
+                type: 'dialog_submission',
+                callback_id: 'dialog_submission_callback_id',
+                channel: {},
+                user: {},
+                team: {},
+              },
+              respond: noop,
+              ack: noop,
+            },
+            { // IncomingEventType.Action for an external_select block (app.options)
+              body: {
+                type: 'block_suggestion',
+                action_id: 'external_select_action_id',
+                channel: {},
+                user: {},
+                team: {},
+                actions: [],
+              },
+              respond: noop,
+              ack: noop,
+            },
+            { // IncomingEventType.Action for "data_source": "external" in dialogs (app.options)
+              body: {
+                type: 'dialog_suggestion',
+                callback_id: 'dialog_suggestion_callback_id',
+                name: 'the name',
+                channel: {},
+                user: {},
+                team: {},
+              },
+              respond: noop,
+              ack: noop,
+            },
+            { // IncomingEventType.ViewSubmitAction (app.view)
+              body: {
+                type: 'view_submission',
+                channel: {},
+                user: {},
+                team: {},
+                view: {
+                  callback_id: 'view_callback_id',
+                }
+              },
+              respond: noop,
+              ack: noop,
+            },
+            // TODO: https://github.com/slackapi/bolt/issues/263
```

Comment:
@shanedewael How about merging this PR before https://github.com/slackapi/bolt/pull/276 and removing these comments out to verify your change works?

Reply:
Yes this is a good idea!